rascal-lsp/src/main/rascal/demo/lang/pico/NewLanguageServer.rsc

```diff
@@ -0,0 +1,187 @@
+@license{
+Copyright (c) 2018-2023, NWO-I CWI and Swat.engineering
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+1. Redistributions of source code must retain the above copyright notice,
+this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright notice,
+this list of conditions and the following disclaimer in the documentation
+and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+}
+module demo::lang::pico::NewLanguageServer
```

Comment:
Nit: Would it be possible to rename the old version to `...::LanguageServerDeprecatedNames` (or something along those line) and call the new version just `...::LanguageServer`?

Reply:
Such a renaming would not be backward compatible.

Reply:
Otherwise the tests explode and also take too long imho

Reply:
We really need the backwards compatibility layer to be tested at every push still; and that layer utilizes the new API so we have full test coverage with the current situation.

Reply:
> Such a renaming would not be backward compatible.

We can update the test to use the old files, but if anyone finds the module (like in a link from somewhere) it would be nice if it's the newest description of how you should write down the file. 

Keeping around an outdated example might confuse more people, especially if it's not clearly marked as outdated.

Reply:
I agree, but backwards compatibility is more important right now. We can clean up after we clean up the deprecated API. Just wait one release. This module is marked @deprecated with a link to the new module right at the top.

Reply:
Don't forget that there are users of the old specific module, people have based their course material on this stuff.

Reply:
Can you elaborate on the backwards compatibility of a demo module? My understanding is that should/can import this module in their rascal code? I've always seen this as documentation, that we happen to also use in our UI tests.

Reply:
We've asked all course owners at four universities; they've no problem with the proposed changes. This is the last thing to do before we merge (and the tests work)

Reply:
done